### ISSUE TITLE
Custom `win_pkg` module does not always get loaded?

### ISSUE THREAD

**Issue by quantonganh** [saltstack/salt#32601]
### Description of Issue/Question

I have a custom `win_pkg` module to fix issue #32571.
I'm sure that it was synced to the minion when running `saltutil.sync_all saltenv='windows'`:

```
     Volume in drive C is Windows 10
     Volume Serial Number is CA31-1C92

     Directory of c:\salt\var\cache\salt\minion\extmods\modules

    04/14/2016  06:52 PM    <DIR>          .
    04/14/2016  06:52 PM    <DIR>          ..
    04/14/2016  06:52 PM            39,165 win_pkg.py
```

The problem is I got this error when install OpenVPN:

```
----------
          ID: openvpn
    Function: pkg.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "C:\salt\bin\lib\site-packages\salt\state.py", line 1626, in call
                  **cdata['kwargs'])
                File "C:\salt\bin\lib\site-packages\salt\loader.py", line 1492, in wrapper
                  return f(*args, **kwargs)
                File "C:\salt\bin\lib\site-packages\salt\states\pkg.py", line 969, in installed
                  **kwargs)
                File "C:\salt\bin\lib\site-packages\salt\states\pkg.py", line 315, in _find_install_targets
                  pkginfo = _get_package_info(name)
                File "C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1072, in _get_package_info
                  return get_repo_data().get('repo', {}).get(name, {})
                File "C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1040, in get_repo_data
                  if exc.errno == errno.ENOENT:
              NameError: global name 'errno' is not defined
     Started: 17:03:39.761000
    Duration: 126.0 ms
     Changes:
```

Pay attention to these lines:

```
                File "C:\salt\bin\lib\site-packages\salt\states\pkg.py", line 315, in _find_install_targets
                  pkginfo = _get_package_info(name)
                File "C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1072, in _get_package_info
                  return get_repo_data().get('repo', {}).get(name, {})
```

Why does it load the builtin module (`C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py`) instead of the custom one in the `extmods`?

The different is I added an additional param to the `_get_package_info()` function:

```
def _get_package_info(name, saltenv='base'):
    '''
    Return package info.
    Returns empty map if package not available
    TODO: Add option for version
    '''
    return get_repo_data(saltenv).get('repo', {}).get(name, {})
```

The strange thing is if I tried to remove that param from the `get_repo_data()` function:

```
-def get_repo_data(saltenv='base'):
+def get_repo_data():
```

then it gets loaded:

```
----------
          ID: openvpn
    Function: pkg.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "C:\salt\bin\lib\site-packages\salt\state.py", line 1626, in call
                  **cdata['kwargs'])
                File "C:\salt\bin\lib\site-packages\salt\loader.py", line 1492, in wrapper
                  return f(*args, **kwargs)
                File "C:\salt\bin\lib\site-packages\salt\states\pkg.py", line 969, in installed
                  **kwargs)
                File "C:\salt\bin\lib\site-packages\salt\states\pkg.py", line 294, in _find_install_targets
                  cur_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True, **kwargs)
                File "c:\salt\var\cache\salt\minion\extmods\modules\win_pkg.py", line 245, in list_pkgs
                  name_map = _get_name_map(saltenv)
                File "c:\salt\var\cache\salt\minion\extmods\modules\win_pkg.py", line 1095, in _get_name_map
                  name_map = get_repo_data(saltenv).get('name_map', {})
              TypeError: get_repo_data() takes no arguments (1 given)
     Started: 19:17:13.708000
    Duration: 16.0 ms
     Changes:
```
### Setup

```
openvpn:
  file:
    - managed
    - name: '{{ opts['cachedir'] }}\openvpn.cer'
    - source: salt://openvpn/client/openvpn.cer
  cmd:
    - wait
    - cwd: {{ opts['cachedir'] }}
    - name: certutil -addstore "TrustedPublisher" openvpn.cer
    - watch:
      - file: openvpn
  pkg:
    - installed
    - require:
      - cmd: openvpn
  service:
    - running
    - name: OpenVPNService
    - enable: True
    - watch:
      - pkg: openvpn
```

with a custom module from this PR: https://github.com/saltstack/salt/pull/32581
### Steps to Reproduce Issue

The master config:

```
file_roots:
  base:
    - /srv/salt
  windows:
    - /path/to/windows

winrepo_dir: /path/to/windows/win/repo
winrepo_dir_ng: /path/to/windows/win/repo-ng
```
- `salt-run winrepo.update_git_repos`
- `salt -G 'os:windows' pkg.refresh_db` saltenv='windows'`
- `saltutil.sync_all saltenv='windows'`
- `state.sls openvpn.client saltenv='windows' -l debug`
### Versions Report

```
Salt Version:
           Salt: 2015.8.1

Dependency Versions:
         Jinja2: 2.8
       M2Crypto: 0.22
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.8 (default, Aug 24 2014, 21:26:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.9.2
       cherrypy: 5.1.0
       dateutil: 2.2
          gitdb: 0.6.4
      gitpython: 1.0.2
          ioflo: Not Installed
        libnacl: 1.4.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.13
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.9.0
        timelib: Not Installed

System Versions:
           dist:
        machine: x86_64
        release: 14.5.0
         system: 10.10.5 -- x86_64
```

and the minion:

```
    ----------
    saltversion:
        2015.8.8
```

**Comment by jfindlay** [comment]
@quantonganh, thanks for reporting.  If you move the default `win_pkg.py` module out of the salt codebase (`C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py`), does it load the external module?  Does it load the external module if you put it into the salt codebase?  You may also want to take care to remove the corresponding `.py?` files if you perform these tests.

**Comment by jfindlay** [comment]
Also loader messages are debug and trace level.  Do you see anything interesting being logged by the loader regarding the `win_pkg` execution modules?

**Comment by quantonganh** [comment]
> If you move the default win_pkg.py module out of the salt codebase (C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py), does it load the external module?

Yes.

> Does it load the external module if you put it into the salt codebase?

Yes.

Another similar error:

```
          ID: 7zip
    Function: pkg.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "c:\salt\bin\lib\site-packages\salt\state.py", line 1626, in call
                  **cdata['kwargs'])
                File "c:\salt\bin\lib\site-packages\salt\loader.py", line 1492, in wrapper
                  return f(*args, **kwargs)
                File "c:\salt\var\cache\salt\minion\extmods\states\pkg.py", line 1029, in installed
                  **kwargs)
                File "c:\salt\var\cache\salt\minion\extmods\states\pkg.py", line 320, in _find_install_targets
                  pkginfo = _get_package_info(name, saltenv=kwargs['saltenv'])
              TypeError: _get_package_info() got an unexpected keyword argument 'saltenv'
     Started: 13:52:11.560000
    Duration: 125.0 ms
     Changes:
```

https://github.com/saltstack/salt/blob/develop/salt/modules/win_pkg.py#L1164

> Do you see anything interesting being logged by the loader on the minion regarding the win_pkg execution modules during any of these tests?

I saw that:

```
    2016-04-19 13:52:12,170 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] pkg_name OpenVPN 2.3.10-I603 : 2.3.10-I603
    2016-04-19 13:52:12,185 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] pkg_name TAP-Windows 9.21.1: 9.21.1
    2016-04-19 13:52:12,185 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] pkg_name Intel(R) Processor Graphics: 20.19.15.4331
    2016-04-19 13:52:12,185 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] pkg_name Notepad++: 6.7.4
    2016-04-19 13:52:12,185 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] key: npp_x86
    2016-04-19 13:52:12,185 [salt.loaded.ext.module.win_pkg][DEBUG   ][2052] pkg_name Go Programming Language amd64 go1.6: 1.6
    2016-04-19 13:52:12,185 [salt.state       ][ERROR   ][2052] An exception occurred in this state: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\state.py", line 1626, in call
        **cdata['kwargs'])
      File "c:\salt\bin\lib\site-packages\salt\loader.py", line 1492, in wrapper
        return f(*args, **kwargs)
      File "c:\salt\var\cache\salt\minion\extmods\states\pkg.py", line 1029, in installed
        **kwargs)
      File "c:\salt\var\cache\salt\minion\extmods\states\pkg.py", line 320, in _find_install_targets
        pkginfo = _get_package_info(name, saltenv=kwargs['saltenv'])
    TypeError: _get_package_info() got an unexpected keyword argument 'saltenv'
```

Why the external module was loaded but it said "unexpected keyword argument 'saltenv'"?

@jfindlay: Do you have any opinion about this?

**Comment by jfindlay** [comment]
@quantonganh, there are basically two circumstances that I know of where a module will fail to load: if the `__virtual__` returns false with an optional reason string, or if there is a syntax error processing the module's code.  I'm not sure whether or how this is related to either of those, but clearly the problem is internal to the pkg state module, and it looks to be fixed in the [latest 2015.8 version](https://github.com/saltstack/salt/blob/v2015.8.8.2/salt/states/pkg.py#L315).
